### PR TITLE
Improve `Factor` typing

### PR DESCRIPTION
**Pull request by danerwilliams**
## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

#### PR Dependency Tree

* **PR #927** 👈

This tree was auto-generated by [Charcoal](https://github.com/danerwilliams/charcoal)

### CODE REVIEW

File: src/mfa/interfaces/factor.interface.ts [hunk 1]
```diff
@@ -11,7 +11,16 @@ export interface Factor {
   type: FactorType;
   sms?: Sms;
   totp?: Totp;
-  userId?: string;
+}
+
+export interface UserManagementFactor {
```
Comment (amygdalama) [inline src/mfa/interfaces/factor.interface.ts:16]: move to `user-management` ?

Comment (amygdalama) [reply src/mfa/interfaces/factor.interface.ts:16]: same w/ the deserializer etc

File: src/user-management/user-management.ts [hunk 2]
```diff
@@ -48,9 +48,13 @@ import {
   Challenge,
   ChallengeResponse,
   Factor,
```
Comment (amygdalama) [inline src/user-management/user-management.ts:50]: does this still need to use the MFA standalone Factor?

Comment (danerwilliams) [reply src/user-management/user-management.ts]: ah yeah had some lingering usage, fixed now

File: src/mfa/interfaces/factor.interface.ts (hunk 1 above)
Comment (danerwilliams) [reply src/mfa/interfaces/factor.interface.ts]: whoops actually created the files in wrong folder, fixed now

File: src/mfa/interfaces/totp.interface.ts [hunk 3]
```diff
@@ -1,28 +1,15 @@
-interface TotpObject {
+export interface Totp {
   issuer: string;
   user: string;
-  qrCode: string;
-  secret: string;
-  uri: string;
+  qrCode?: string;
```
Comment (amygdalama) [inline src/mfa/interfaces/totp.interface.ts:4]: are these optional or required for standalone MFA responses?

Comment (danerwilliams) [reply src/mfa/interfaces/totp.interface.ts:4]: for both standalone and userland mfa, the secret properties (qr code, secret, and uri) are used in some cases and not used in others. 

so they should be optional for both

Comment (amygdalama) [reply src/mfa/interfaces/totp.interface.ts:4]: ah ok, so seems like the standalone MFA code for retrieving a totp factor was broken? 🤔  (asking for context so i know what to put in the changelog)

File: src/user-management/interfaces/user-management-factor.interface.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { Totp, TotpResponse } from '../../mfa/interfaces/totp.interface';
+
+export interface UserManagementFactor {
```
Comment (amygdalama) [inline src/user-management/interfaces/user-management-factor.interface.ts:3]: now that these are in `user-management`, what do you think about dropping the `UserManagement` prefix from the names?

File: src/mfa/interfaces/totp.interface.ts (hunk 3 above)
Comment (danerwilliams) [reply src/mfa/interfaces/totp.interface.ts:4]: yes! the standalone mfa api was inacurrately saying that there was always access to these responses even in methods that didn't have them

File: src/user-management/interfaces/user-management-factor.interface.ts (hunk 4 above)
Comment (danerwilliams) [reply src/user-management/interfaces/user-management-factor.interface.ts:3]: yeah that seems reasonable

Comment (danerwilliams) [reply src/user-management/interfaces/user-management-factor.interface.ts]: done

**Review by amygdalama — APPROVED** [review]

**Pull request merged**